internal/tools/aws/database_multitenant.go

```diff
@@ -0,0 +1,637 @@
+	}
+
+	lockedCluster, err := d.lockRDSCluster(*vpc.VpcId, store, logger)
+	if err != nil {
+		return errors.Wrapf(err, "unable to lock a RDS cluster for installation ID %s", d.installationID)
+	}
+	defer lockedCluster.unlock(logger)
+
+	masterSecretValue, err := d.client.Service().secretsManager.GetSecretValue(&secretsmanager.GetSecretValueInput{
+		SecretId: lockedCluster.cluster.DBClusterIdentifier,
+	})
+	if err != nil && !IsErrorCode(err, secretsmanager.ErrCodeResourceNotFoundException) {
+		return errors.Wrapf(err, "unable to find the master secret for RDS cluster ID %s", *lockedCluster.cluster.DBClusterIdentifier)
+	}
+
+	close, err := d.connectRDSCluster(rdsMySQLSchemaInformationDatabase, *lockedCluster.cluster.Endpoint, DefaultMattermostDatabaseUsername, *masterSecretValue.SecretString)
+	defer close(logger)
+	if err != nil {
+		return errors.Wrapf(err, "unable to connect to RDS cluster ID %s", *lockedCluster.cluster.DBClusterIdentifier)
+	}
+
+	ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(DefaultMySQLDefaultContextTimeout*time.Second))
+	defer cancel()
+
+	err = d.createDatabaseIfNotExist(ctx, databaseName)
+	if err != nil {
+		return errors.Wrapf(err, "unable to create schema in RDS cluster ID %s", *lockedCluster.cluster.DBClusterIdentifier)
+	}
+
+	installationSecreteName := RDSMultitenantSecretName(d.installationID)
+
+	installationSecretValue, err := d.client.Service().secretsManager.GetSecretValue(&secretsmanager.GetSecretValueInput{
+		SecretId: aws.String(installationSecreteName),
+	})
+	if err != nil && !IsErrorCode(err, secretsmanager.ErrCodeResourceNotFoundException) {
+		return errors.Wrapf(err, "failed to get RDS database secret for installation ID %s", d.installationID)
+	}
+
+	var installationSecret *RDSSecret
+	if installationSecretValue != nil && installationSecretValue.SecretString != nil {
+		installationSecret, err = unmarshalSecretPayload(*installationSecretValue.SecretString)
+		if err != nil {
+			return errors.Wrapf(err, "failed to unmarshal RDS database secret for installation ID %s", d.installationID)
+		}
+	} else {
+		description := RDSMultitenantClusterSecretDescription(d.installationID, *lockedCluster.cluster.DBClusterIdentifier)
+		tags := []*secretsmanager.Tag{
+			{
+				Key:   aws.String(trimTagPrefix(DefaultRDSMultitenantDBClusterIDTagKey)),
+				Value: lockedCluster.cluster.DBClusterIdentifier,
+			},
+			{
+				Key:   aws.String(trimTagPrefix(DefaultRDSMultitenantVPCIDTagKey)),
+				Value: aws.String(*vpc.VpcId),
+			},
+			{
+				Key:   aws.String(trimTagPrefix(DefaultMattermostInstallationIDTagKey)),
+				Value: aws.String(d.installationID),
+			},
+		}
+		installationSecret, err = d.createInstallationSecret(installationSecreteName, d.installationID, description, tags)
+		if err != nil {
+			return errors.Wrapf(err, "failed to create a RDS database secret for installation ID %s", d.installationID)
+		}
+	}
```

Comment:
This section is probably another good candidate for a helper method. A comment at the top of the stanza would be helpful at the least.